### ISSUE TITLE
Release 1.3.1

### ISSUE THREAD

**Issue by yanniszark** [kubeflow/manifests#1890]
### Problem Statement

The 1.3.0 release is out and KubeCon is over! This means we should start working on 1.3.1.

I would like to ask each WG to let me know of items you want to include in 1.3.1, by commenting on this issue. Please keep in mind that this is a point release, so we should include bugfixes, not new features.

1. **Upstream WGs:** resolve blocking PRs **Deadline: June 21st**
    - Manifests:
        - [x] Upgrade cert-manager https://github.com/kubeflow/manifests/issues/1838
        - [x] Upgrade to latest Istio 1.9.x https://github.com/kubeflow/manifests/issues/1893
        - [x] Upgrade Knative https://github.com/kubeflow/manifests/issues/1754
    - WG-Pipelines:
        - [x] Introduce token credential inferface: https://github.com/kubeflow/pipelines/pull/5287
        - [x] KFP Tekton: Upgrade to `v0.8.0`
    - WG-AutoML: No new changes (source: @johnugeorge)
    - WG-Notebooks:
        - [x] Fix Jupyter Web App using workspace mountpath set in the configuration: kubeflow/kubeflow#5952
        - [x] Fix GPU allocation for notebooks: kubeflow/kubeflow#5891
        - [x] Fix KNative Istio permissions in profile controller: kubeflow/kubeflow#5848
        - [x] Fix profile namespace labels for Istio: kubeflow/kubeflow#5761
        - [ ] Remove resource quota from namespace if removed from profile: kubeflow/kubeflow#5950
        - [ ] Fix notebook-controller not filtering objects properly: kubeflow/kubeflow#5873
        - [ ] Fix missing parameter in central dashboard: kubeflow/kubeflow#5782
        - [ ] Fix email regex being too restrictive: kubeflow/kubeflow#5679
        - [ ] Fix bug in kfam due to using an outdated version of the profile API: kubeflow/kubeflow#5941
        - [ ] Central Dashboard manifests doesn't use configmap: kubeflow/kubeflow#5958
        - [ ] KFAM doesn't create Istio authorization policy for namespace contributors kubeflow/kubeflow#5984
2. **Upstream WGs:** release artifacts (images, manifests), give manifests-wg the desired git revisions 
3. **Manifests WG:** sync manifests from upstream app releases to manifests repo 
4. **Manifests WG:** Cut release candidate branch 
5. **Manifests WG:** test and update instructions.  
6. **Distribution Owners:** Use instructions from wg-manifests for how to use kustomizations and update their distributions. 

cc @kubeflow/wg-automl-leads
cc @kubeflow/wg-deployment-leads
cc @kubeflow/wg-manifests-leads
cc @kubeflow/wg-notebooks-leads
cc @kubeflow/wg-pipeline-leads
cc @kubeflow/wg-serving-leads
cc @kubeflow/wg-training-leads

**Comment by thesuperzapper** [comment]
@kubeflow/wg-manifests-leads I think an important feature for `1.3.1` for Manifests WG would be to add the "supported version matrix" for external dependancies that kubeflow uses, like "cert-manager", and "knative"

See issue: https://github.com/kubeflow/manifests/issues/1842

**Comment by DavidSpek** [comment]
There are serious security vulnerabilities in Istio and Envoy that allowed for people to bypass authorization. The Istio manifests should be updated to at least the latest version of 1.9. Updating to 1.10 would also be a good option, as the only thing incompatible with Istio 1.10 seems to be [an existing problem](https://github.com/kubeflow/kubeflow/pull/5925) in the tensorboard-controller that is [solved by adding the `--bind_all` flag to the startup command](https://github.com/kubeflow/kubeflow/pull/5925).

**Comment by juliusvonkohout** [comment]
> There are serious security vulnerabilities in Istio and Envoy that allowed for people to bypass authorization. The Istio manifests should be updated to at least the latest version of 1.9. Updating to 1.10 would also be a good option, as the only thing incompatible with Istio 1.10 seems to be [an existing problem](https://github.com/kubeflow/kubeflow/pull/5925) in the tensorboard-controller that is [solved by adding the `--bind_all` flag to the startup command](https://github.com/kubeflow/kubeflow/pull/5925).

If you upgrade istio please consider switching from the ratelimited dockerhub to the official istio gcr  repository.

```
istioctl manifest generate --set hub='gcr.io/istio-release'
# --set global.proxy.resources.requests=50m might also be interesting
```

I am using this with more parameters (e.g.  --set tag='1.9.2')  for Minikube and Openshift.

**Comment by DavidSpek** [comment]
@yanniszark I just created the issue https://github.com/kubeflow/manifests/issues/1893, although it isn't very difficult to find as it is all stated in the Istio release notes. There are 7 vulnerabilities that have been patched since 1.9.0, 6 of which have CVSS scores ranging from 7.5 to 10. I'd also to note that it should be the norm to update software dependencies, and not only updated out of absolute necessity or serious CVEs. Not keeping up with updates just causes more work in the long run, especially if transition periods are missed.

**Comment by DavidSpek** [comment]
PRs/issues for WG-Notebooks that should to be addressed for 1.3.1

Remove resource quota from namespace if removed from profile: https://github.com/kubeflow/kubeflow/issues/5950

Fix Jupyter Web App using workspace mountpath set in the configuration: https://github.com/kubeflow/kubeflow/pull/5952
Fix GPU allocation for notebooks: https://github.com/kubeflow/kubeflow/pull/5891
Fix notebook-controller not filtering objects properly: https://github.com/kubeflow/kubeflow/pull/5873
Fix KNative Istio permissions in profile controller: https://github.com/kubeflow/kubeflow/pull/5848
Fix missing parameter in central dashboard: https://github.com/kubeflow/kubeflow/pull/5782
Fix profile namespace labels for Istio: https://github.com/kubeflow/kubeflow/pull/5761
Fix email regex being too restrictive: https://github.com/kubeflow/kubeflow/pull/5679

**Comment by johnugeorge** [comment]
Also, kfp package in notebook images are outdated- 1.0.4.  Can new package be added to image once client changes for https://github.com/kubeflow/pipelines/issues/5138 are added to sdk ?

**Comment by DavidSpek** [comment]
@johnugeorge Thanks for reminding me, there are actually a lot more updates that need to happen to the notebook images. I'll create a PR to update them tomorrow.

**Comment by DavidSpek** [comment]
Another issue from the Notebook Working Group:
Central Dashboard manifests doesn't use configmap: https://github.com/kubeflow/kubeflow/issues/5958

**Comment by andreyvelich** [comment]
As we discussed earlier, we will release Sobol and Restart Strategy support for Katib in Kubeflow 1.4

For Kubeflow 1.3.1 we will cut new Katib tag: v0.11.1 with the latest fixes for 0.11 branch.
cc @kubeflow/wg-automl-leads

**Comment by DavidSpek** [comment]
@yanniszark For WG Notebooks the notebook server images also should be updated. I'm already halfway through this process, which will be another 2 PRs (1 is already created, last one needs the existing one to be merged first).

**Comment by yanniszark** [comment]
> @yanniszark For WG Notebooks the notebook server images also should be updated. I'm already halfway through this process, which will be another 2 PRs (1 is already created, last one needs the existing one to be merged first).

@DavidSpek thanks. Can you make an issue to track, as for the other features?

**Comment by johnugeorge** [comment]
@DavidSpek client credentials PR  https://github.com/kubeflow/pipelines/pull/5676 needs to be merged to get multi tenant working in pipeline. Can we wait to have a new Pipeline SDK release and then get it in the new notebook image?

**Comment by zijianjoy** [comment]
cc @Bobgy

**Comment by pvaneck** [comment]
Want to make sure that the kfp-tekton component update isn't forgotten for the 1.3.1 release (updating from `v0.8.0-rc0` to `v0.8.0`)
`Update KFP Tekton release with bugfixes: https://github.com/kubeflow/kfp-tekton/commits/v0.8.0 `

**Comment by yuzisun** [comment]
For Serving WG, we do not have plan to upgrade the KFServing version, but would like to get with this [PR](https://github.com/kubeflow/manifests/pull/1768) in to upgrade KNative to 0.17.

**Comment by thesuperzapper** [comment]
I am leaving a link to https://github.com/kubeflow/website/issues/2658 which is about updating component docs for 1.3 (some components are still on 1.2 docs).

**Comment by DavidSpek** [comment]
@yanniszark  https://github.com/kubeflow/kubeflow/pull/5848 has been merged and can be marked as done

**Comment by DavidSpek** [comment]
@yanniszark Got an urgent issue for the notebooks working group that needs to be added.

Profile controller doesn't create Istio authorization policy for namespace contributors: https://github.com/kubeflow/kubeflow/issues/5984

**Comment by DavidSpek** [comment]
@johnugeorge The notebook server images have been updated as well as the manifests for the Jupyter Web App (this was done in https://github.com/kubeflow/kubeflow/pull/5998).

**Comment by johnugeorge** [comment]
Thanks @DavidSpek. But default kfp package still won't work within notebook as PR is not merged yet?

**Comment by DavidSpek** [comment]
@johnugeorge I'm not sure what you mean. All the packages in the notebook server images were updated and the manifests we updated with those images, so they will be the default included with 1.3.1. You can also change the Jupyter Web App configmap in your existing deployment to include the new images if you would like to use them.

**Comment by johnugeorge** [comment]
@DavidSpek I was referring to https://github.com/kubeflow/pipelines/pull/5676

**Comment by DavidSpek** [comment]
@yanniszark Was the release for 1.3.1 already cut? I we might need to do an urgent upgrade to Istio 1.9.6 since there was a “ remotely exploitable vulnerability where credentials specified in the Gateway and DestinationRule credentialName field can be accessed from different namespaces” that has been solved. 

See the [ISTIO-SECURITY-2021-007](https://istio.io/latest/news/security/istio-security-2021-007/) bulletin for more details.

CVSS Score: 9.1 [CVSS:3.1/AV:N/AC:L/PR:L/UI:N/S:C/C:H/I:L/A:L](https://www.first.org/cvss/calculator/3.1#CVSS:3.1/AV:N/AC:L/PR:L/UI:N/S:C/C:H/I:L/A:L)

Release notes: https://istio.io/latest/news/releases/1.9.x/announcing-1.9.6/

**Comment by thesuperzapper** [comment]
@yanniszark @DavidSpek this is a strong reason not to bundle external dependancies in our manifests, but rather describe a way for users to install them directly from the upstream source. For example, pointing them to the upstream method of install, like a helm chart.

Obviously, we need to set bounds on what versions (of things like istio) are supported for a specific Kubeflow release, but, it's pretty simple to just say, we support all versions between 1.6.X-1.9.X.

**Comment by DavidSpek** [comment]
@thesuperzapper If we tell people to install the Istio operator (with helm or istioctl), we can just provide a simple `Istio-spec.yaml` that defines the required Istio configuration that users can deploy to install Istio. This can then be very easily kept up to date with Renovate. See https://github.com/argoflow/argoflow-aws/pull/148 for example.

**Comment by juliusvonkohout** [comment]
> @thesuperzapper If we tell people to install the Istio operator (with helm or istioctl), we can just provide a simple `Istio-spec.yaml` that defines the required Istio configuration that users can deploy to install Istio. This can then be very easily kept up to date with Renovate. See [argoflow/argoflow-aws#148](https://github.com/argoflow/argoflow-aws/pull/148) for example.

This would also make it very easy to provide specs for different platforms like plain kubernetes or openshift or provide specs with rootless istio-cni

**Comment by yanniszark** [comment]
@DavidSpek thanks for the heads-up.
I have opened issue https://github.com/kubeflow/manifests/issues/1924 to track the istio upgrade to 1.9.6.
We will move forward with cutting the first RC and testing.
In parallel, we will work to include 1.9.6, which shouldn't include many changes.
Let's continue the discussion there.

**Comment by pvaneck** [comment]
If possible this release, this small knative patch would be worthwhile to have so that users can list  knative resources in their namespaces: https://github.com/kubeflow/manifests/pull/1877

**Comment by kimwnasptd** [comment]
> If possible this release, this small knative patch would be worthwhile to have so that users can list knative resources in their namespaces: #1877

@pvaneck I merged your referenced PR, thanks for mentioning it. I've also opened a PR to cherry-pick this for 1.3.1 https://github.com/kubeflow/manifests/pull/1948. I'll also create a new rc tag once it gets merged

**Comment by kimwnasptd** [comment]
Just a heads up, I've created the `1.3.1-rc.1` tag to include #1948 as well.

The distribution testing phases has also started and you can track it's progress here https://github.com/kubeflow/manifests/issues/1949

**Comment by juliusvonkohout** [comment]
@davidspek i encountered the bug from https://github.com/kubeflow/kubeflow/issues/5982 that prevents you from deleting notebooks using the webinterface. Notebbook deletion hangs on the finalizer then. Do you think that is relevant for 1.3.1?

**Comment by kimwnasptd** [comment]
@juliusvonkohout thanks for reporting it, let's try to investigate what's the problem in this issue. Once we have a more clear picture of the underlying problem we can evaluate if/how to handle it for 1.3.1

**Comment by akdigitalself** [comment]
Istio 1.9 is reaching end of life on August 18th: https://istio.io/latest/news/support/announcing-1.9-eol/ 
Will Istio 1.10 be supported in 1.3.1? Thanks!

**Comment by DavidSpek** [comment]
@akdigitalself I’ve been using Istio 1.10 since it was released without any problems so it will be supported by 1.3.1.

**Comment by juliusvonkohout** [comment]
> @juliusvonkohout thanks for reporting it, let's try to investigate what's the problem in this issue https://github.com/kubeflow/kubeflow/issues/5982. Once we have a more clear picture of the underlying problem we can evaluate if/how to handle it for 1.3.1

Alright, I  have provided all the information there and someone has already created a pull request.

**Comment by kimwnasptd** [comment]
closing this issue since we've release 1.3.1
